dynatrace/environment_v2/problems.py

```diff
@@ -0,0 +1,301 @@
+        self.affected_entities: Optional[List[EntityStub]] = [EntityStub(raw_element=e) for e in raw_element.get("affectedEntities", [])]
+        self.recent_comments: Optional[CommentList] = CommentList(raw_element=raw_element.get("recentComments"))
+        self.impacted_entities: Optional[List[EntityStub]] = [EntityStub(raw_element=e) for e in raw_element.get("impactedEntities", [])]
+        self.linked_problem_info: Optional[LinkedProblem] = LinkedProblem(raw_element=raw_element.get("linkedProblemInfo"))
+        self.root_cause_entity: Optional[EntityStub] = EntityStub(raw_element=raw_element.get("rootCauseEntity"))
+        self.problem_filters: Optional[List[AlertingProfileStub]] = [AlertingProfileStub(raw_element=a) for a in raw_element.get("problemFilters", [])]
+        self.evidence_details: Optional[EvidenceDetails] = EvidenceDetails(raw_element=raw_element.get("evidenceDetails"))
+        self.impact_analysis: Optional[ImpactAnalysis] = ImpactAnalysis(raw_element=raw_element.get("impactAnalysis"))
+        self.entity_tags: Optional[List[METag]] = [METag(raw_element=t) for t in raw_element.get("entityTags", [])]
+
+
+class ProblemCloseResult(DynatraceObject):
+    def _create_from_raw_data(self, raw_element: Dict[str, Any]):
+        self.problem_id: str = raw_element.get("problemId")
+        self.closing: bool = raw_element.get("closing")
+        self.close_timestamp: datetime = int64_to_datetime(raw_element.get("closeTimestamp"))
+        self.comment: Optional[Comment] = Comment(raw_element=raw_element.get("comment"))
+
+
+class LinkedProblem(DynatraceObject):
+    def _create_from_raw_data(self, raw_element: Dict[str, Any]):
+        self.display_id: str = raw_element.get("displayId")
+        self.problem_id: str = raw_element.get("problemId")
+
+
+class CommentList(DynatraceObject):
+    def _create_from_raw_data(self, raw_element: Dict[str, Any]):
+        self.total_count: int = raw_element.get("totalCount")
+        self.comments: Optional[List[Comment]] = [Comment(raw_element=c) for c in raw_element.get("comments", [])]
+
+
+class Comment(DynatraceObject):
+    def _create_from_raw_data(self, raw_element: Dict[str, Any]):
+        self.created_at: datetime = int64_to_datetime(raw_element.get("createdAtTimestamp"))
+        self.author: Optional[str] = raw_element.get("authorName")
+        self.context: Optional[str] = raw_element.get("context")
+        self.id: Optional[str] = raw_element.get("id")
+        self.content: Optional[str] = raw_element.get("content")
+
+
+class EvidenceDetails(DynatraceObject):
+    def _create_from_raw_data(self, raw_element: Dict[str, Any]):
+        self.total_count: int = raw_element.get("totalCount")
+        self.details: Optional[List[Evidence]] = (
+            [EventEvidence(raw_element=e) for e in raw_element.get("details", []) if e.get("evidenceType") == EvidenceType.EVENT]
```

Comment:
You need to change `e.get("evidenceType")` to `EvidenceType(e.get("evidenceType"))` , otherwise this will always be false and no evidences are being created (the array is empty). This is why checking the len fails in the test

Reply:
Sorry! I totally missed that :) I'm actually inclined to just add .value to all the right sides (e.g. `EvidenceType.EVENT.value`) instead. Seems a bit tidier. What do you think?

Reply:
Also this highlighted a blindspot in my test. For some reason the len did not fail on my machine